src/http/router.rs

```diff
@@ -154,9 +154,38 @@ pub type Params = Captures<'static, 'static>;
 /// # fn handle_single_segment(req: Request, params: Params) -> anyhow::Result<Response> { todo!() }
 /// # fn handle_exact(req: Request, params: Params) -> anyhow::Result<Response> { todo!() }
 /// ```
+///
+/// Route based on the trailing segment of a Spin wildcard route, instead of on the full path
+///
+/// ```no_run
+/// // spin.toml
+/// //
+/// // [[trigger.http]]
+/// // route = "/shop/..."
+///
+/// // component
+/// # use spin_sdk::http::{IntoResponse, Params, Request, Response, Router};
+/// fn handle_route(req: Request) -> Response {
+///     let mut router = Router::suffix();
+///     router.any("/users/*", handle_users);
+///     router.any("/products/*", handle_products);
+///     router.handle(req)
+/// }
+///
+/// // '/shop/users/1' is routed to `handle_users`
+/// // '/shop/products/1' is routed to `handle_products`
+/// # fn handle_users(req: Request, params: Params) -> anyhow::Result<Response> { todo!() }
+/// # fn handle_products(req: Request, params: Params) -> anyhow::Result<Response> { todo!() }
+/// ```
 pub struct Router {
     methods_map: HashMap<Method, MethodRouter<Box<dyn Handler>>>,
     any_methods: MethodRouter<Box<dyn Handler>>,
+    route_on: RouteOn,
+}
+
+enum RouteOn {
```

Comment:
I know it's internal but can you add some short doc comments on each one of these variants?